nixos.nix

```
@@ -90,13 +90,67 @@ in
         # Create a directory in persistent storage, so we can bind
         # mount it.
         mkDirCreationSnippet = persistentStoragePath: dir:
-          let
-            targetDir = concatPaths [ persistentStoragePath dir ];
-          in
           ''
-            if [[ ! -e "${targetDir}" ]]; then
-                mkdir -p "${targetDir}"
-            fi
+            # Given a source directory, /source, and a target directory,
+            # (/target/foo/bar/bazz, we want to "clone" the target structure
+            # from source into the target. Essentially, we want to get both
+            # /source/target/foo/bar/bazz and /target/foo/bar/bazz to exist
+            # in the filesystem. More concretely, we'd like to map
```

I think you missed this question from my last suggestion (easy to miss with that giant wall of text suggestion):

"we want to get both /source/target/foo/bar/bazz [1] and /target/foo/bar/bazz [2] to exist" -- shouldn't [1] already exist, if we're wanting to map [1] to [2]?

---

While I'm here:

```suggestion
            # from source into the target. Essentially, we want both
            # /source/target/foo/bar/bazz and /target/foo/bar/bazz to exist
            # on the filesystem. More concretely, we'd like to map
```